tests/Feature/Engine/MessageEndEventTest.php

```diff
@@ -0,0 +1,197 @@
+        $messageFlow->setCollaboration($collaboration);
+        $messageFlow->setSource($messageEndEventB);
+        $messageFlow->setTarget($eventA);
+        $collaboration->addMessageFlow($messageFlow);
+
+        $eventA = $processA->getEvents()->item(1);
+        $eventB = $processB->getEvents()->item(1);
+
+        $dataStoreA = $this->dataStoreRepository->createDataStoreInstance();
+        $dataStoreA->putData('A', '1');
+
+        $dataStoreB = $this->dataStoreRepository->createDataStoreInstance();
+        $dataStoreB->putData('B', '1');
+
+        $dataStoreCollectionA = new DataStoreCollection();
+        $dataStoreCollectionA->add($dataStoreA);
+
+        $dataStoreCollectionB = new DataStoreCollection();
+        $dataStoreCollectionB->add($dataStoreB);
+
+        $processA->setDataStores($dataStoreCollectionA);
+        $processB->setDataStores($dataStoreCollectionB);
+
+        $instanceA = $this->engine->createExecutionInstance($processA, $dataStoreA);
+        $instanceB = $this->engine->createExecutionInstance($processB, $dataStoreB);
+
+        // we start the second process and run it up to the end
+        $startB = $processB->getEvents()->item(0);
+        $activityB1 = $processB->getActivities()->item(0);
+
+        // we start the process A
+        $startA = $processA->getEvents()->item(0);
+        $activityA1 = $processA->getActivities()->item(0);
+
+        $startA->start();
+        $this->engine->runToNextState();
+
+        //Assertion: The activity must be activated
+        $this->assertEvents([
+            EventInterface::EVENT_EVENT_TRIGGERED,
+            ActivityInterface::EVENT_ACTIVITY_ACTIVATED,
+        ]);
+
+        // we finish the first activity so that the catch message is activated
+        $tokenA = $activityA1->getTokens($instanceA)->item(0);
+        $activityA1->complete($tokenA);
+        $this->engine->runToNextState();
+
+        $this->assertEvents([
+            ActivityInterface::EVENT_ACTIVITY_COMPLETED,
+            ActivityInterface::EVENT_ACTIVITY_CLOSED,
+            IntermediateCatchEventInterface::EVENT_CATCH_TOKEN_ARRIVES
+        ]);
+
+        $startB->start();
+        $this->engine->runToNextState();
+
+        //Assertion: Process B - The activity must be activated
+        $this->assertEvents([
+            EventInterface::EVENT_EVENT_TRIGGERED,
+            ActivityInterface::EVENT_ACTIVITY_ACTIVATED,
+        ]);
+
+        $tokenB = $activityB1->getTokens($instanceB)->item(0);
+        $activityB1->complete($tokenB);
+        $this->engine->runToNextState();
+
+        //Assertion: Process B - The activity is completed and the end event must be activated
+        $this->assertEvents([
+            ActivityInterface::EVENT_ACTIVITY_COMPLETED,
+            ActivityInterface::EVENT_ACTIVITY_CLOSED,
+
+            // the throw token of the end is sent
+            IntermediateThrowEventInterface::EVENT_THROW_TOKEN_ARRIVES,
+
```

Comment:
Missing events for the Message sent.
Ex.
MessageEventInterface::EVENT_THROW_EVENT
IntermediateThrowEventInterface:: EVENT_EVENT_THROW